src/python/zquantum/qaoa/ansatzes/xz_ansatz.py

```diff
@@ -0,0 +1,122 @@
+from zquantum.core.interfaces.ansatz import Ansatz, ansatz_property
+
+from zquantum.core.circuits import Circuit
+from zquantum.core.evolution import time_evolution
+from zquantum.core.openfermion import change_operator_type
+
+from openfermion import QubitOperator, IsingOperator
+from openfermion.utils import count_qubits
+from typing import Union, Optional, List
+import numpy as np
+import sympy
+from overrides import overrides
+from itertools import combinations
+
+from zquantum.qaoa.ansatzes.x_ansatz import ncr
+
+
+class QAOAXZAnsatz(Ansatz):
+
+    supports_parametrized_circuits = True
+    cost_hamiltonian = ansatz_property("cost_hamiltonian")
+    type = ansatz_property("type")
+
+    def __init__(
+        self,
+        number_of_layers: int,
+        cost_hamiltonian: Union[QubitOperator, IsingOperator],
+        type: int,
+    ):
+        """This is implementation of the XZ Ansatzes from https://arxiv.org/abs/2105.01114 section 4.2
+
+        Args:
+            number_of_layers: k-body depth (the maximum number of qubits entangled at one time) as described in https://arxiv.org/abs/2105.01114.  Cannot be greater than the number of qubits.
+            cost_hamiltonian: Hamiltonian representing the cost function
+            type: either 0 or 1, from the two types of XZ ansatzes from https://arxiv.org/abs/2105.01114
```

Comment:
I would try to think about a better way to describe it than `type`. 
Perhas `use_k_body_z_operators` which would be a boolean?